internal/service/certificatesigningrequest.go

```diff
@@ -268,42 +258,48 @@ func (h *ServiceHandler) ReplaceCertificateSigningRequest(ctx context.Context, r
 func (h *ServiceHandler) ApproveCertificateSigningRequest(ctx context.Context, request server.ApproveCertificateSigningRequestRequestObject) (server.ApproveCertificateSigningRequestResponseObject, error) {
 	orgId := store.NullOrgId
 
-	csr, err := h.store.CertificateSigningRequest().Get(ctx, orgId, request.Name)
+	storeCsr := h.store.CertificateSigningRequest()
+	csr, err := storeCsr.Get(ctx, orgId, request.Name)
 	if err != nil {
 		if errors.Is(err, flterrors.ErrResourceNotFound) {
 			return server.ApproveCertificateSigningRequest404JSONResponse{}, nil
-		} else {
-			return nil, err
 		}
+		return nil, err
 	}
 
 	if api.IsStatusConditionTrue(csr.Status.Conditions, api.CertificateSigningRequestDenied) {
 		return server.ApproveCertificateSigningRequest409JSONResponse{Message: "The request has already been denied"}, nil
 	}
+	if api.IsStatusConditionTrue(csr.Status.Conditions, api.CertificateSigningRequestApproved) {
+		return server.ApproveCertificateSigningRequest409JSONResponse{Message: "The request has already been approved"}, nil
+	}
+
+	signedCert, err := signApprovedCertificateSigningRequest(h.ca, csr)
+	if err != nil {
+		return nil, err
+	}
 
 	approvedCondition := api.Condition{
 		Type:    api.CertificateSigningRequestApproved,
 		Status:  api.ConditionStatusTrue,
 		Reason:  "Approved",
 		Message: "Approved",
 	}
-	err = h.store.CertificateSigningRequest().UpdateConditions(ctx, orgId, request.Name, []api.Condition{approvedCondition})
-	if err != nil {
-		if errors.Is(err, flterrors.ErrResourceNotFound) {
-			return server.ApproveCertificateSigningRequest404JSONResponse{}, nil
-		}
-		return nil, err
-	}
 
-	err = signApprovedCertificateSigningRequest(h.ca, csr)
-	if err != nil {
-		if errors.Is(err, flterrors.ErrResourceNotFound) {
-			return server.ApproveCertificateSigningRequest404JSONResponse{}, nil
-		}
+	csr.Status.Certificate = &signedCert
+	csr.Status.Conditions = append(csr.Status.Conditions, approvedCondition)
+	_, err = storeCsr.Update(ctx, orgId, csr)
+
+	switch err {
+	case nil:
+		return server.ApproveCertificateSigningRequest200JSONResponse{}, nil
+	case flterrors.ErrResourceNotFound:
+		return server.ApproveCertificateSigningRequest404JSONResponse{Message: "Resource not found"}, nil
+	case flterrors.ErrNoRowsUpdated, flterrors.ErrResourceVersionConflict:
+		return server.ApproveCertificateSigningRequest409JSONResponse{Message: "Version conflict or no rows updated"}, nil
```

Comment:
Maybe you can add the text from the errors themselves? There are 2 possible errors here

Reply:
Sure that's easy